test/integration/vorpal-engine_test.go

```diff
@@ -34,26 +39,84 @@ func TestScanVorpal_SentFileWithoutExtension_FailCommandWithError(t *testing.T)
+	assert.NilError(t, err, "Sending empty source file should not fail")
 }
 
-func TestExecuteVorpalScan_VorpalLatestVersionSetFalse_SuccessfullyReturnMockData(t *testing.T) {
+func TestExecuteVorpalScan_VorpalLatestVersionSetFalse_Success(t *testing.T) {
+	configuration.LoadConfiguration()
+	args := []string{
+		"scan", "vorpal",
+		flag(commonParams.SourcesFlag), "data/python-vul-file.py",
+		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
+	}
 
-	scanResult, _ := commands.ExecuteVorpalScan("data/python-vul-file.py", false)
-	expectedMockResult := commands.ReturnFailureResponseMock()
-	//TODO: update mocks when there's a real engine
-	assert.DeepEqual(t, scanResult, expectedMockResult)
+	err, _ := executeCommand(t, args...)
+	assert.NilError(t, err, fmt.Sprintf("Should not fail with error: %v", err))
 }
 
 func TestExecuteVorpalScan_CorrectFlagsSent_SuccessfullyReturnMockData(t *testing.T) {
+	configuration.LoadConfiguration()
+	args := []string{
+		"scan", "vorpal",
+		flag(commonParams.SourcesFlag), "data/python-vul-file.py",
+		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
+	}
+
+	err, _ := executeCommand(t, args...)
+	assert.NilError(t, err, fmt.Sprintf("Should not fail with error: %v", err))
+}
+
+func TestExecuteVorpalScan_InitializeAndRun_Success(t *testing.T) {
+	configuration.LoadConfiguration()
+	args := []string{
+		"scan", "vorpal",
+		flag(commonParams.SourcesFlag), "",
+		flag(commonParams.VorpalLatestVersion),
+		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
+	}
+
+	vorpalWrapper := grpcs.NewVorpalGrpcWrapper(viper.GetInt(commonParams.VorpalPortKey))
+	if healthCheckErr := vorpalWrapper.HealthCheck(); healthCheckErr == nil {
+		_ = vorpalWrapper.ShutDown()
+	} else {
+		t.Failed()
+	}
+
+	err, _ := executeCommand(t, args...)
+	assert.NilError(t, err, "Sending empty source file should not fail")
+
+	args = []string{
+		"scan", "vorpal",
+		flag(commonParams.SourcesFlag), "data/python-vul-file.py",
+		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
+	}
+
+	err, _ = executeCommand(t, args...)
+	assert.NilError(t, err, "Scan should not fail with error")
+}
+
+func TestExecuteVorpalScan_UnsupportedLanguage_Fail(t *testing.T) {
+	configuration.LoadConfiguration()
+	args := []string{
+		"scan", "vorpal",
+		flag(commonParams.SourcesFlag), "data/positive1.tf",
+		flag(commonParams.AgentFlag), commonParams.DefaultAgent,
+	}
 
-	scanResult, _ := commands.ExecuteVorpalScan("data/python-vul-file.py", true)
-	expectedMockResult := commands.ReturnSuccessfulResponseMock()
-	//TODO: update mocks when there's a real engine
-	assert.DeepEqual(t, scanResult, expectedMockResult)
+	err, bytes := executeCommand(t, args...)
+	assert.NilError(t, err, "Scan should not fail with error")
+	var scanResult grpcs.ScanResult
+	err = json.Unmarshal(bytes.Bytes(), &scanResult)
+	assert.NilError(t, err, "Failed to unmarshal scan result")
+	assert.Equal(t, scanResult.Error.Description, "The file extension is not supported.")
```

Comment:
is this string "The file extension is not supported." on our side?
if no- you have to check only that we got an error and no more